sui_programmability/examples/games/sources/drand_random_selection.move

```diff
@@ -0,0 +1,215 @@
+///   round N). The randomness of round X can be retrieved using
+///  `curl https://drand.cloudflare.com/8990e7a9aaed2ffed73dbd7092123d6f289930540d7651336225dc172e51b2ce/public/X'.
+/// - Users can join the game as long as it is not closed and receive a *ticket*.
+/// - Anyone can *complete* the game by proving drand's randomness of round N, which is used to declare the winner.
+/// - The owner of the winning "ticket" can request a "winner ticket" and finish the game.
+/// As long as someone is closing the game in time (or at least before round N) we have the guarantee that the winner is
+/// selected using unpredictable and unbiasable randomness. Otherwise, someone could wait until the randomness of round N
+/// is public, see if it could win the game and if so, join the game and drive it to completion. Therefore, honest users
+/// are encourged to close the game in time.
+///
+/// All the external inputs needed for the following APIs can be retrieved from one of drand's public APIs, e.g. using
+/// the above curl commands.
+///
+module games::drand_random_selection {
+    use std::vector;
+
+    use sui::object::{Self, ID, UID};
+    use std::option::{Self, Option};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+    use sui::bls12381;
+    use std::hash::sha2_256;
+
+    /// Error codes
+    const EGameNotInProgress: u64 = 0;
+    const EGameAlreadyCompleted: u64 = 1;
+    const EInvalidRandomness: u64 = 2;
+    const EInvalidSeed: u64 = 3;
+    const EInvalidTicket: u64 = 4;
+
+    /// Game status
+    const IN_PROGRESS: u8 = 0;
+    const CLOSED: u8 = 1;
+    const COMPLETED: u8 = 2;
+
+    /// Game represents a set of parameters of a single game.
+    /// This game can be extended to require ticket purchase, reward winners, etc.
+    ///
+    struct Game has key, store {
+        id: UID,
+        round: u64,
+        status: u8,
+        participants: u64,
+        winner: Option<u64>,
+    }
+
+    /// Ticket represents a participant in a single game.
+    /// Can be deconstructed only by the owner.
+    struct Ticket has key, store {
+        id: UID,
+        game_id: ID,
+        participant_index: u64,
+    }
+
+    /// GameWinner represents a participant that won in a specific game.
+    /// Can be deconstructed only by the owner.
+    struct GameWinner has key, store {
+        id: UID,
+        game_id: ID,
+    }
+
+    /// Create a shared-object Game.
+    public entry fun create(round: u64, ctx: &mut TxContext) {
+        let game = Game {
+            id: object::new(ctx),
+            round,
+            status: IN_PROGRESS,
+            participants: 0,
+            winner: option::none(),
+        };
+        transfer::share_object(game);
+    }
+
+    /// Anyone can close the game by providing the randomness of round-2.
+    public entry fun close(game: &mut Game, drand_sig: vector<u8>, drand_prev_sig: vector<u8>) {
```

Comment:
How about using the unchained version and document it properly? I think it's easier and cheaper (in gas and tx size) to demonstrate at this stage.

Reply:
The public api currently only serve the chained mode

Reply:
Pfff.. it's a pity but ok fair enough.